### PR TITLE
Add GRC config policy metric in 2.6

### PR DESCRIPTION
**Pull request by oafischer**
https://github.com/stolostron/backlog/issues/25514

@gparvin Wasn't sure if a new topic made sense in governance_metric.adoc or not, since that would have required the overview topic name and link and I'm trying to avoid something like that with the link issues we had recently. Still, I'll ask my team and see what they say.

### CODE REVIEW

File: governance/governance_metric.adoc [hunk 1]
```diff
@@ -24,3 +24,14 @@ These labels and values enable queries that can show us many things happening in
 
 *Note:* If the metrics are not needed, and there are any concerns about performance or security, this feature can be disabled. Set the `DISABLE_REPORT_METRICS` environment variable to `true` in the propagator deployment. You can also add `policy_governance_info` metric to the observability allowlist as a custom metric. See link:../observability/customize_observability.adoc#adding-custom-metrics[Adding custom metrics] for more details.
 
+The `config_policies_evaluation_duration_` histogram tracks the number of seconds it takes to process all configuration policies on the cluster. Use the following metrics to implement the histogram:
```
Comment (mprahl) [inline governance/governance_metric.adoc]: Could this be in a new subsection?

Comment (mprahl) [inline governance/governance_metric.adoc]:
```suggestion
The `config_policies_evaluation_duration_` histogram tracks the number of seconds it takes to process all configuration policies that are ready to be evaluated on the cluster. Use the following metrics to query the histogram:
```

Comment (oafischer) [reply governance/governance_metric.adoc]: Absolutely, I was thinking about that. If we make it a new subsection, we also need to either rebrand the overview or make `policy_governance_info` its own subsection as well. Do you have any preference on that?

File: governance/governance_metric.adoc [hunk 2]
```diff
@@ -24,3 +24,14 @@ These labels and values enable queries that can show us many things happening in
...
+
+- `config_policies_evaluation_duration_seconds_bucket`: The buckets are cumulative and represent seconds with the following possible entries: 1, 3, 9, 10.5, 15, 30, 90, 120, 300, 450, 600, and above.
+- `config_policies_evaluation_duration_seconds_count`: The count of all events.
+- `config_policies_evaluation_duration_seconds_sum`: The sum of all values.
+
+Use `config_policies_evaluation_duration_` to change the `ConfigurationPolicy` `evaluationInterval` setting when the time to process policies exceeds the default interval of ten seconds. To receive information about the time used to evaluate configuration policies, use a metric that resembles the following expression:
```
Comment (mprahl) [inline governance/governance_metric.adoc]: This first sentence isn't quite right. This information can be used to determine if the `ConfigurationPolicy` `spec.evaluationInterval` should be adjusted on resource intensive policies that don't require as frequent evaluation. Alternatively, the concurrency (a link here would be good) can be increased at the cost of more resource utilization on the Kubernetes API server.

File: governance/governance_metric.adoc (hunk 1 above)
Comment (mprahl) [reply governance/governance_metric.adoc]: I prefer the latter option.

**Comment by oafischer** [comment]
Thanks @mprahl, I've implemented your suggestions. @dockerymick do the titles look OK? Not sure if we should use colons or not. I also wasn't sure how to handle the variable metric since it has 3 options at the end.

**Comment by gparvin** [comment]
Thanks, this looks good to me.  Holding for others to take a look too.
/hold
/lgtm

File: governance/governance_metric.adoc [hunk 3]
```diff
@@ -24,3 +29,19 @@ These labels and values enable queries that can show us many things happening in
 
 *Note:* If the metrics are not needed, and there are any concerns about performance or security, this feature can be disabled. Set the `DISABLE_REPORT_METRICS` environment variable to `true` in the propagator deployment. You can also add `policy_governance_info` metric to the observability allowlist as a custom metric. See link:../observability/customize_observability.adoc#adding-custom-metrics[Adding custom metrics] for more details.
 
+[#metric-config-policies-evaluation-duration-]
+=== Metric: config_policies_evaluation_duration_
+
+The `config_policies_evaluation_duration_` histogram tracks the number of seconds it takes to process all configuration policies that are ready to be evaluated on the cluster. Use the following metrics to query the histogram:
+
+- `config_policies_evaluation_duration_seconds_bucket`: The buckets are cumulative and represent seconds with the following possible entries: 1, 3, 9, 10.5, 15, 30, 90, 120, 300, 450, 600, and above.
```
Comment (dockerymick) [inline governance/governance_metric.adoc]: remove "above"; change that to **greater**

File: governance/governance_metric.adoc [hunk 4]
```diff
@@ -24,3 +29,19 @@ These labels and values enable queries that can show us many things happening in
...
+- `config_policies_evaluation_duration_seconds_count`: The count of all events.
+- `config_policies_evaluation_duration_seconds_sum`: The sum of all values.
+
+Use `config_policies_evaluation_duration_` to determine if the `ConfigurationPolicy` `evaluationInterval` setting needs to be changed for resource intensive policies that do not need frequent evaluation. You can also increase the concurrency at the cost of higher resource utilization on the Kubernetes API server. See xref:../config_policy_ctrl.adoc##configure-config-policy-controller[Configure the configuration policy controller] for more details.
```
Comment (dockerymick) [inline governance/governance_metric.adoc]: extra hashtag being used in the link reference

File: observability/observe_environments.adoc [hunk 5]
```diff
@@ -62,6 +62,21 @@ View the following table of metric types that are supported by the framework:
...
+| Stable. See link:../governance/governance_metric.adoc#gov-metric[Governance metric] for more details.
+
+| `config_policies_evaluation_duration_seconds_count`
+| Histogram
+| 
+| Stable. See link:../governance/governance_metric.adoc#gov-metric[Governance metric] for more details.
+
+| `config_policies_evaluation_duration_seconds_sum`
+| Histogram
+|
```
Comment (dockerymick) [inline observability/observe_environments.adoc]: Not sure what labels and tags should be used. If it's not applicable, we might add a dash ( - ). i am checking the style guide to verify

**Review by dockerymick — CHANGES_REQUESTED** [review]
left a couple of comments

File: observability/observe_environments.adoc (hunk 5 above)
Comment (oafischer) [reply observability/observe_environments.adoc]: Saw that you asked Brandi and will use None like she suggested, thanks for looking into that!

**Comment by oafischer** [comment]
Made some changes, please approve if it looks good now @dockerymick @gparvin @mprahl @bjoydeep. Thanks!

File: governance/governance_metric.adoc [hunk 6]
```diff
@@ -24,3 +29,19 @@ These labels and values enable queries that can show us many things happening in
 
 *Note:* If the metrics are not needed, and there are any concerns about performance or security, this feature can be disabled. Set the `DISABLE_REPORT_METRICS` environment variable to `true` in the propagator deployment. You can also add `policy_governance_info` metric to the observability allowlist as a custom metric. See link:../observability/customize_observability.adoc#adding-custom-metrics[Adding custom metrics] for more details.
 
+[#metric-config-policies-evaluation-duration-]
+=== Metric: config_policies_evaluation_duration_
+
+The `config_policies_evaluation_duration_` histogram tracks the number of seconds it takes to process all configuration policies that are ready to be evaluated on the cluster. Use the following metrics to query the histogram:
+
+- `config_policies_evaluation_duration_seconds_bucket`: The buckets are cumulative and represent seconds with the following possible entries: 1, 3, 9, 10.5, 15, 30, 90, 120, 300, 450, 600, and greater.
```
Comment (mprahl) [inline governance/governance_metric.adoc]:
```suggestion
- `config_policies_evaluation_duration_seconds_bucket`: The buckets are cumulative and represent seconds with the following possible entries: 1, 3, 9, 10.5, 15, 30, 60, 90, 120, 180, 300, 450, 600, and greater.
```

File: governance/governance_metric.adoc [hunk 7]
```diff
@@ -24,3 +29,19 @@ These labels and values enable queries that can show us many things happening in
...
+- `config_policies_evaluation_duration_seconds_count`: The count of all events.
+- `config_policies_evaluation_duration_seconds_sum`: The sum of all values.
+
+Use `config_policies_evaluation_duration_` to determine if the `ConfigurationPolicy` `evaluationInterval` setting needs to be changed for resource intensive policies that do not need frequent evaluation. You can also increase the concurrency at the cost of higher resource utilization on the Kubernetes API server. See xref:../config_policy_ctrl.adoc#configure-config-policy-controller[Configure the configuration policy controller] for more details.
+
+To receive information about the time used to evaluate configuration policies, use a metric that resembles the following expression:
```
Comment (mprahl) [inline governance/governance_metric.adoc]:
```suggestion
To receive information about the time used to evaluate configuration policies, perform a Prometheus query that resembles the following expression:
```

**Review by mprahl — APPROVED** [review]
/hold for other reviews

This looks great!

**Review by dockerymick — APPROVED** [review]
Looks great Oliver! Just a heads up for the future, when there are more than 2 sections on the page, consider creating a table of contents for the page

**Review by gparvin — APPROVED** [review]
Looks good!

**Pull request merged**